xapian-letor/tests/api_letor.cc

```diff
@@ -94,26 +167,127 @@ DEFINE_TESTCASE(createfeaturevector, generated)
+    TEST_EQUAL(mset.size(), 2);
     fv = fl.create_feature_vectors(mset, Xapian::Query("lions"), db);
     TEST_EQUAL(fv.size(), 2);
     TEST_EQUAL(fv[0].get_fcount(), 19);
     TEST_EQUAL(fv[1].get_fcount(), 19);
+    return true;
+}
+
+DEFINE_TESTCASE(createfeaturevectoronevector, generated)
+{
+    Xapian::FeatureList fl;
+    Xapian::Database db = get_database("apitest_ranker2",
+				       db_index_one_document);
+    Xapian::Enquire enquire(db);
+    enquire.set_query(Xapian::Query("tigers"));
+    Xapian::MSet mset;
+    vector<Xapian::FeatureVector> fv =
+	    fl.create_feature_vectors(mset, Xapian::Query("tigers"), db);
+    TEST(fv.empty());
+    mset = enquire.get_mset(0, 10);
+    TEST(!mset.empty());
+    fv = fl.create_feature_vectors(mset, Xapian::Query("tigers"), db);
+    TEST_EQUAL(fv.size(), 1);
+    TEST_EQUAL(fv[0].get_fcount(), 19);
+    return true;
+}
+
+DEFINE_TESTCASE(createfeaturevectoronevector_wrongquery, generated)
+{
+    Xapian::FeatureList fl;
+    Xapian::Database db = get_database("apitest_ranker3",
+				       db_index_one_document);
+    Xapian::Enquire enquire(db);
+    enquire.set_query(Xapian::Query("llamas"));
+    Xapian::MSet mset;
+    vector<Xapian::FeatureVector> fv =
+	    fl.create_feature_vectors(mset, Xapian::Query("llamas"), db);
+    TEST(fv.empty());
+
+    mset = enquire.get_mset(0, 10);
+    TEST(mset.empty());
+    fv = fl.create_feature_vectors(mset, Xapian::Query("llamas"), db);
+    TEST_EQUAL(fv.size(), 0);
+    return true;
+}
+
+DEFINE_TESTCASE(createfeaturevectorthree, generated)
+{
+    Xapian::FeatureList fl;
+    Xapian::Database db = get_database("apitest_ranker4",
+				       db_index_three_documents);
+    Xapian::Enquire enquire(db);
+    enquire.set_query(Xapian::Query("score"));
+    Xapian::MSet mset;
+    vector<Xapian::FeatureVector> fv =
+	    fl.create_feature_vectors(mset, Xapian::Query("score"), db);
+    TEST(fv.empty());
+
+    mset = enquire.get_mset(0, 10);
+    TEST(!mset.empty());
+    fv = fl.create_feature_vectors(mset, Xapian::Query("score"), db);
+    TEST_EQUAL(fv.size(), 2);
+    TEST_EQUAL(fv[0].get_fcount(), 19);
+    TEST_EQUAL(fv[1].get_fcount(), 19);
+    return true;
+}
+
+
+DEFINE_TESTCASE(preparetrainingfileonedb, generated)
+{
+    string db_path = get_database_path("apitest_listnet_ranker1",
+				       db_index_one_document);
+    string data_directory = test_driver::get_srcdir() + "/testdata/";
+    string query = data_directory + "queryone.txt";
+    string qrel = data_directory + "qrelone.txt";
+    string training_data = data_directory + "training_data_one_document.txt";
+    Xapian::prepare_training_file(db_path, query, qrel, 10,
+				  "training_output_data_one_doc.txt");
+    ifstream if1(training_data);
+    ifstream if2("training_output_data_one_doc.txt");
+    string file1;
+    string file2;
+    while (getline(if1, file1)) {
+	TEST(getline(if2, file2));
+	istringstream iss1(file1);
+	istringstream iss2(file2);
```

Comment:
The names `file1` and `file2` are rather misleading - I'd suggest calling these variables `line1` and `line2` since they hold the contents of a line from the corresponding stream.